### PR TITLE
OSD-7084 alert when cert will expire

### PR DESCRIPTION
**Pull request by yithian**
this enables metric scraping by prometheus and includes updates to the local testing tooling

### CODE REVIEW

File: deploy/service.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+apiVersion: v1
```
Comment (dustman9000) [inline deploy/service.yaml:1]: I think we should probably move the Service and ServiceMonitor out into the hack/test/deploy directory instead as this is handled automatically be the operator-sdk/olm

File: hack/test/clusterdeploy.yaml [hunk 2]
```diff
@@ -14,69 +14,40 @@ metadata:
   namespace: certtest
 spec:
   installed: true
-  baseDomain: dnstest.aws.openshift.com
+  baseDomain: dev.yuggoth.space
```
Comment (dustman9000) [inline hack/test/clusterdeploy.yaml:17]: Maybe make generic like app.example.com?

Comment (dustman9000) [reply hack/test/clusterdeploy.yaml:17]: Can we move this file into something like hack/test/deploy?

File: hack/test/local_test.sh [hunk 3]
```diff
@@ -2,7 +2,7 @@
 # This script will test certman-operator in minikube by doing the following:
 # 1. Create a test cluster locally.
 # 2. Install certman's dependencies and CRDs.
-# 3. Create a docker image of the operator.
+# 3. Create a podman image of the operator.
```
Comment (dustman9000) [inline hack/test/local_test.sh:5]:
```suggestion
# 3. Create a podman/docker image of the operator.
```

File: hack/test/local_test.sh [hunk 4]
```diff
@@ -66,23 +66,24 @@ if [[ ! $(pwd) =~ .*certman-operator$ ]]; then
   exit
 fi
 
-echo "Checking if docker service is active"
-systemctl is-active docker
+#echo "Checking if docker service is active"
```
Comment (dustman9000) [inline hack/test/local_test.sh:69]: Can we support both podman and docker in this script?

File: hack/test/local_test.sh [hunk 5]
```diff
@@ -66,23 +66,24 @@ if [[ ! $(pwd) =~ .*certman-operator$ ]]; then
...
+#systemctl is-active docker
 
 testdir="${initial_dir}/hack/test"
 tmpdir="${initial_dir}/hack/test/tmp"
 mkdir ${tmpdir}
 
 # Start minikube with extra memory (needed for the go build)
-minikube start -p certtest --memory='2500mb'
+minikube start -p certtest --memory=6g --bootstrapper=kubeadm --extra-config=kubelet.authentication-token-webhook=true --extra-config=kubelet.authorization-mode=Webhook --extra-config=scheduler.address=0.0.0.0 --extra-config=controller-manager.address=0.0.0.0 --extra-config=apiserver.service-node-port-range=1-65535
```
Comment (dustman9000) [inline hack/test/local_test.sh:77]: Move extra params into a variable to make it more maintainable?

File: hack/test/local_test.sh [hunk 6]
```diff
@@ -66,23 +66,24 @@ if [[ ! $(pwd) =~ .*certman-operator$ ]]; then
...
 kubectl config use-context certtest
 
 # Install openshift router
 cd $tmpdir
 git clone git@github.com:openshift/router.git
 cd router
-kubectl create ns openshift-ingress
-kubectl create -n openshift-ingress -f deploy/
+kubectl create -n openshift-ingress -f deploy/route_crd.yaml
```
Comment (dustman9000) [inline hack/test/local_test.sh:84]: Can we move this into a hack/test/deploy directory?

File: hack/test/namespace.yaml [hunk 7]
```diff
@@ -3,6 +3,8 @@ apiVersion: v1
 kind: Namespace
```
Comment (dustman9000) [inline hack/test/namespace.yaml:3]: ditto into a hack/test/deploy directory

File: hack/test/clusterdeploy.yaml (hunk 2 above)
Comment (yithian) [reply hack/test/clusterdeploy.yaml:17]: ugh, i thought i excluded that change. i'll revert hostname change and move the clusterdeployment into hack/test/deploy.yml

File: hack/test/local_test.sh (hunk 6 above)
Comment (yithian) [reply hack/test/local_test.sh:84]: that specific resource is from the openshift/router git repo, so it's not something we have to worry about

File: hack/test/local_test.sh (hunk 4 above)
Comment (yithian) [reply hack/test/local_test.sh]: that should be added now

**Pull request merged**